lib/WP_Auth0_Options.php

```diff
@@ -1,67 +1,253 @@
 <?php
 
-class WP_Auth0_Options extends WP_Auth0_Options_Generic {
+class WP_Auth0_Options {
 
+	/**
+	 * Name used in options table option_name column.
+	 *
+	 * @var string
+	 */
+	protected $_options_name = 'wp_auth0_settings';
+
+	/**
+	 * Current array of options stored in memory.
+	 *
+	 * @var null|array
+	 */
+	private $_opts = null;
+
+	/**
+	 * Array of options overridden by constants.
+	 *
+	 * @var array
+	 */
+	protected $constant_opts = [];
+
+	/**
+	 * @var WP_Auth0_Options
+	 */
 	protected static $_instance = null;
-	protected $_options_name    = 'wp_auth0_settings';
 
+	/**
+	 * WP_Auth0_Options constructor.
+	 * Finds and stores all constant-defined settings values.
+	 */
+	public function __construct() {
+		$option_keys = $this->get_defaults( true );
+		foreach ( $option_keys as $key ) {
+			$setting_const = $this->get_constant_name( $key );
+			if ( defined( $setting_const ) ) {
+				$this->constant_opts[ $key ] = constant( $setting_const );
+			}
+		}
+	}
+
+	/**
+	 * @return WP_Auth0_Options
+	 */
 	public static function Instance() {
 		if ( null === self::$_instance ) {
-			self::$_instance = new WP_Auth0_Options;
+			self::$_instance = new self;
 		}
 		return self::$_instance;
 	}
 
-	public function is_wp_registration_enabled() {
-		return is_multisite() ? users_can_register_signup_filter() : get_site_option( 'users_can_register' );
+	/**
+	 * Takes an option key and creates the constant name to look for.
+	 *
+	 * @param string $key - Option key to transform.
+	 *
+	 * @return string
+	 */
+	public function get_constant_name( $key ) {
+		// NOTE: the add_filter call must load before WP_Auth0::init() so it cannot be used in a theme.
```

Comment:
should this be moved to the README or some place with more visibility?

Reply:
It's in the docs as well 👍